conans/util/windows.py

```diff
@@ -85,7 +85,7 @@ def path_shortener(path, short_paths):
     # to current user to avoid
     # access problems in cygwin/msys2 windows subsystems when using short_home folder
     try:
-        userdomain, username = os.getenv("USERDOMAIN"), os.environ["USERNAME"]
+        userdomain, username = os.getenv("USERDOMAIN"), os.getenv("USERNAME")
```

Comment:
Failing faster always has my preference. I can revert this commit and catch the `KeyError` 

or we can accept that  the used domain will be `None` in my case  and executing that command `cacls <short_home_path> /E /G "None":F` exits with `Invalid arguments` and has no effect when I run it in Powershell.

But both solutions also mean that the workaround for the `short_paths` in cygwin/msys2 subsystems will not work when both `USERDOMAIN` and `USERNAME` are not set as environmental variables. I could attempt to fix the workaround, if I could reproduce the original behavior, but it might be better to do this in a separate issue and PR. A possible fix for the workaround is using the output of `whoami`?

powershell
```powershell
PS C:\dev> whoami
<domain>\<username>
```
cmd
```cmd
C:\dev>whoami
<domain>\<username>
```

mingw is the odd duck and would need some extra processing. Domain is capitalized and the `\` has been replaced with `+`
```shell
MINGW64 /c/dev
$ whoami
<DOMAIN>+<username>
```

Which approach has your preference @memsharded

Reply:
> Failing faster always has my preference. I can revert this commit and catch the KeyError

Yes, lets do ``KeyError`` and fail fast.
Keep this PR just for that, lets solve this, which is easy and low risk.

Regarding the cygwin/msys2 subsystems I am not that concerned. To be honest, the whole ``short_paths`` thing disappear in 2.0, so as long as there is some workaround, I wouldn't put much effort on this.